ocaml/xapi/message_forwarding.ml

```diff
@@ -4379,5 +4379,24 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
              (fun () ->
                 do_op_on ~__context ~local_fn ~host
                   (fun session_id rpc -> Client.Cluster_host.disable rpc session_id self)))
+
+    let forget ~__context ~self =
+      info "Cluster_host.forget cluster_host:%s" (Ref.string_of self);
+      let cluster = Db.Cluster_host.get_cluster ~__context ~self in
+      let local_fn = Local.Cluster_host.disable ~self in
+      (* We need to ask another host that has a cluster host to mark it as dead.
+       * We might've run force destroy and this host would no longer have a cluster host
+       * *)
+      let other_hosts =
+        Db.Cluster.get_cluster_hosts ~__context ~self:cluster
+        |> List.filter ((<>) self) in
+      match other_hosts with
```

Comment:
What happens if this gets forwarded to one of the dead hosts, or one in `pending_forget`, is that possible?
Should we keep trying until we find a live one?

Reply:
Yes we need to find a live host somehow, we need to retry if we get a HOST_OFFLINE error.